pkg/kube/kubehttp_test.go

```diff
@@ -0,0 +1,129 @@
+package kube_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	wego "github.com/weaveworks/weave-gitops/api/v1alpha1"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+var testClustername = "test-cluster"
+var testenv *envtest.Environment
+var cfg *rest.Config
+var testclient client.Client
+var err error
+var scheme *apiruntime.Scheme
+var k kube.Kube
+var ns corev1.Namespace
+
+var _ = BeforeSuite(func() {
+	testenv = &envtest.Environment{CRDDirectoryPaths: []string{"../../manifests/crds"}}
+
+	scheme = kube.CreateScheme()
+	cfg, err = testenv.Start()
+
+	Expect(err).NotTo(HaveOccurred())
+
+	ns = corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: "wego-system"},
+	}
+
+	testclient, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	err = testclient.Create(context.Background(), &ns)
+	Expect(err).NotTo(HaveOccurred())
+})
+
+var _ = BeforeEach(func() {
+	k = &kube.KubeHTTP{
```

Comment:
wondering if it would be ok to move this the first line of the next `Describe` 🤔

Reply:
I had originally wanted a clean environment on each test, but the `testenv` library doesn't work that way. This object is currently stateless, so not much value in creating a new one every time, but if it has state in the future, I want a clean state on each test.

Reply:
that makes sense to me 👍🏼